drivers/HwWatchdog.cpp

```diff
@@ -0,0 +1,100 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "HwWatchdog.h"
+
+static LowPowerTicker ticker;//tiker for periodic delay call back for kick()
+static bool is_watchdog_started = false; //boolean to control watchdog start and stop
+#define MS_TO_US(x) (x * 1000) //macro to convert millisecond to microsecond 
+
+/** Refreshes the watchdog timer.
+ *
+ * This function should be called periodically before the watchdog times out.
+ * Otherwise, the system is reset.
+ *
+ * If the watchdog timer is not currently running this function does nothing
+ */
+static void kick()
+{
+    core_util_critical_section_enter();
+    hal_watchdog_kick();
+    core_util_critical_section_exit();
+}
+
+uint32_t max_timeout()
+{
+    const watchdog_features_t features = hal_watchdog_get_platform_features();
+    return features.max_timeout;
+}
+
+mbed_error_status_t start()
+{
+
+    watchdog_status_t sts;
+    mbed_error_status_t msts = MBED_SUCCESS;
+    MBED_STATIC_ASSERT(HW_WATCHDOG_TIMEOUT >= 0,"Timeout must be greater than zero");
+    MBED_ASSERT(HW_WATCHDOG_TIMEOUT < max_timeout());
+    if(is_watchdog_started)
```

Comment:
You may want to do this check inside the critical section to prevent possible race conditions with two threads calling start for the first time at the same time.